week4/week4.playground/Contents.swift

```diff
@@ -0,0 +1,203 @@
+Create an array of Int? called numsWithNil with the following values:
+79, nil, 80, nil, 90, nil, 100, 72
+
+Create an array of Int called numsBy2 with values starting at 2 through 100, by 2.
+Create an array of Int called numsBy4 with values starting at 2 through 100, by 4.
+
+You can set the values of the arrays above using whatever method you find the easiest. In previous weeks you were introduced to ranges and sequences in Swift. Leveraging those in the Array initializer will allow you to create the requested arrays in a single line. Don’t let the last two break your stride.
+
+- Create a function called evenNumbersArray that takes a parameter of [Int] (array of Int) and returns [Int]. The array of Int returned should contain all the even numbers in the array passed. Call the function passing the nums array and print the output.
+
+- Create a function called sumOfArray that takes a parameter of [Int?] and returns an Int. The function should return the sum of the array values passed that are not nil. Call the function passing the numsWithNil array, and print out the results.
+
+- Create a function called commonElementsSet that takes two parameters of [Int] and returns a Set<Int> (set of Int). The function will return a Set<Int> of the values in both arrays.
+
+- Call the function commonElementsSet passing the arrays numsBy2, numsBy4, and print out the results.
+*/
+
+let nums = Array(0...100)
+let numsWithNil: [Int?] = [79, nil, 80, nil, 90, nil, 100, 72]
+
+var numsBy2 = [Int]()
+for number in stride(from: 2, to: 100, by: 2) {
+  numsBy2.append(number)
+}
+print("numsBy2 array will be \(numsBy2)")
+
+var numsBy4 = [Int]()
+for number in stride(from: 2, to: 100, by: 4) {
+  numsBy4.append(number)
+}
+print("numsBy4 array will be \(numsBy4)")
+
+func evenNumbersArray(numberArray: [Int]) -> [Int] {
+  var evenNumbers = numberArray.filter { $0 % 2 == 0 }
+  return evenNumbers
+}
+
+print(evenNumbersArray(numberArray: nums))
+
+func sumOfArray(optionalArray: [Int?]) -> Int {
+  var sumOfNonNilValues = optionalArray.compactMap { Int($0 ?? 0)}.reduce(0, +)
+  return sumOfNonNilValues
+}
+
+print("sum of non nil values is \(sumOfArray(optionalArray: numsWithNil))")
+
+func commonElementsSet(numArrayOne: [Int], numArrayTwo: [Int]) -> Set<Int> {
+  let combinedArray = numArrayOne + numArrayTwo
+  let resultSet = Set(combinedArray)
+  return resultSet
+}
+
+print("Set of two arrays \(commonElementsSet(numArrayOne: numsBy2, numArrayTwo: numsBy4))")
+
+
+// d) Create a struct called Square that has a stored property called sideLength and a computed property called area. Create an instance of Square and print the area.
+
+struct Square {
+  var sideLength: Int
+  var area: Int {
+    sideLength * sideLength
+  }
+}
+
+let mySquare = Square(sideLength: 10)
+
+print("the area of mySquare is \(mySquare.area)")
+
+// Above and beyond
+protocol Shape {
+  func calculateArea() -> Double
+}
+
+struct Circle: Shape {
+  var radius: Double
+  private var pi = 3.14
```

Comment:
No need to create a `private` var called `pi`, there's one already available in the library, just call it with `.pi`. You also don't need the initializer in that case.